app/src/main/java/org/p2p/wallet/bridge/send/ui/BridgeSendPresenter.kt

```diff
@@ -272,15 +277,54 @@ class BridgeSendPresenter(
         if (oldInputAmount != inputAmount && calculationMode.getCurrencyMode() !is CurrencyMode.Fiat) {
             updateInputValue(inputAmount, true)
         }
-        view?.setMaxButtonVisible(calculationMode.inputAmountDecimal.isZero())
+        val feeTotalInToken = getFeeTotalInToken()
+        val validAmount = token.token.total - feeTotalInToken > BigDecimal.ZERO
+        view?.setMaxButtonVisible(calculationMode.inputAmountDecimal.isZero() && validAmount)
+    }
+
+    private fun BridgeSendContract.View.handleUpdateTotal(sendFee: SendFee?) {
+        calculationMode.isCurrentInputEmpty()
+        val amount = calculationMode.inputAmountDecimal.orZero()
+        val total = countTotal(amount, sendFee)
+        val totalFormatted = if (amount.isZero()) {
+            emptyString()
+        } else {
+            total.formatWithDecimals(calculationMode.getCurrencyMode().fractionLength)
+        }
+        setTotalValue(totalFormatted)
+    }
+
+    private fun countTotal(amount: BigDecimal, sendFee: SendFee?): BigDecimal {
+        val bridgeFee = sendFee as? SendFee.Bridge
+        val fees = getFeeList(bridgeFee)
+        val fee = if (calculationMode.getCurrencyMode() is CurrencyMode.Fiat.Usd) {
+            fees.sumOf {
+                it.amountInUsd.toBigDecimalOrZero()
+            }
+        } else {
+            fees.sumOf {
+                it.amountInToken
```

Comment:
one liner here would be ok
fees.sumOf { it.amountInToken } or fees.sumOf(Fee::amountInToken)